fastapi/routing.py

```diff
@@ -119,6 +121,18 @@ def _prepare_response_content(
     return res
 
 
+def _merge_lifespan_context(
+    original_context: Lifespan[Any], nested_context: Lifespan[Any]
+) -> Lifespan[Any]:
+    @asynccontextmanager
+    async def merged_lifespan(app: AppType) -> AsyncIterator[Mapping[str, Any]]:
+        async with original_context(app) as maybe_self_context:
+            async with nested_context(app) as maybe_nested_context:
+                yield {**(maybe_self_context or {}), **(maybe_nested_context or {})}
```

Comment:
The _yielded_ value is actually called `maybe_state` on Starlette's code. For consistency, let's keep the same name.
 
```suggestion
        async with original_context(app) as maybe_original_state:
            async with nested_context(app) as maybe_nested_state:
                yield {**(maybe_original_state or {}), **(maybe_nested_state or {})}
```